# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1423282514 18000
#      Fri Feb 06 23:15:14 2015 -0500
# Node ID 409d82472aee386c887851cfad95bf91c064a088
# Parent  ff56a98991017f2f20a6562c046cf6dd9d566f38
Disable output flow control in the unix terminal with Ctrl-Q/S (bug #42681)

* kpty.cpp (KPty::open): Disable output flow control in the unix terminal.
Ctrl-S no longer appears to freeze the command window, and Ctrl-Q and Ctrl-S
are now available for readline key bindings.

diff --git a/libgui/qterminal/libqterminal/unix/kpty.cpp b/libgui/qterminal/libqterminal/unix/kpty.cpp
--- a/libgui/qterminal/libqterminal/unix/kpty.cpp
+++ b/libgui/qterminal/libqterminal/unix/kpty.cpp
@@ -381,16 +381,17 @@ bool KPty::open()
 #endif
 
 #endif /* HAVE_OPENPTY */
   fcntl(d->masterFd, F_SETFD, FD_CLOEXEC);
   fcntl(d->slaveFd, F_SETFD, FD_CLOEXEC);
 
   struct ::termios t;
   tcGetAttr(&t);
+  t.c_iflag &= ~IXON;
   t.c_lflag &= ~ECHOCTL;
   tcSetAttr(&t);
   return true;
 }
 
 void KPty::closeSlave()
 {
     Q_D(KPty);
